src/Razor/Microsoft.AspNetCore.Mvc.Razor.Extensions/src/ViewComponentTagHelperDescriptorFactory.cs

```diff
@@ -172,7 +172,7 @@ private void AddRequiredAttributes(ImmutableArray<IParameterSymbol> methodParame
         {
             foreach (var parameter in methodParameters)
             {
-                if (GetIndexerValueTypeName(parameter) == null)
+                if (GetIndexerValueTypeName(parameter) == null && parameter.HasExplicitDefaultValue != true)
```

Comment:
nit: Should add a comment explaining the default value here given consumers on Razor's end may not fully understand Roslyn's semantic here